sunpy/net/dataretriever/sources/gong.py

```diff
@@ -90,13 +93,19 @@ def _get_url_for_timerange(self, timerange, **kwargs):
                 if not wavelength_in:
                     patterns.append(url_pattern_1), patterns.append(url_pattern_2)
                 else:
-                    wave = int(kwargs['wavelength'].min.value)
-                    if (6562 <= wave <= 6563):
-                        patterns.append(url_pattern_2)
-                    elif wave == 6768:
-                        patterns.append(url_pattern_1)
-
-
+                    try:
+                        wave_float = kwargs['wavelength'].min.value
+                        da = []
+                        da.append(wave_float)
+                        for wave_nums in pattern_table.keys():
+                            db = []
+                            db.append(wave_nums)
+                            if np.isclose(da, db, 1e-10, 1e-10):
+                                wave = wave_nums
```

Comment:
This has appeared already before... should you make a function somewhere (maybe utils) and then use it from everywhere?